### PR TITLE
New Extension: Nativebase Documentation

### PR DESCRIPTION
**Pull request by ShubhamVerma1811**
## Description

This extensions helps to search the nativebase.io documentation from within the Raycast. It also support different versions of Nativebase documentations.

## Screencast

ss-versions
ss-pages

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

**Comment by ShubhamVerma1811** [comment]
Found a bug, fixing it.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @ShubhamVerma1811 👋 

Thanks for your contribution 💯 What does the most of the users do here? Are they using different versions at the same time?

Could you move everything related to the README to a folder in the root called `media` instead, icons used in the extension is in the right folder 💪 

 There are a few approaches that could fit:
- The way it is now, but it makes an extra step before its “ready” to use
- Create a preference dropdown with the versions (so only one version is in use)
- Create each version as a separate commands (So the users can “disable” and even bind hotkeys and alias)

Could one of the last two fit in instead? 

Another approach about the rate limit could be to cache the result for an hour, this way it's possible to remove the GitHub Personal Token part completely!

Let me know what you think 😄 Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by ShubhamVerma1811** [comment]
Hey @pernielsentikaer, Thank you for the feedback!
- I could move README assets in `media`
-  Also look into caching the results

> Create a preference dropdown with the versions (so only one version is in use)

I personally feel like as a user I would switch between the versions for checking the docs if something was updated or added in the docs that was not documented in other versions. And having a set preference version won't be helping with that.

> Create each version as a separate commands (So the users can “disable” and even bind hotkeys and alias)

I am unclear what you meant here. Did you something like this?
If not, Could you please share some example of what you meant?

Screenshot 2022-03-06 at 16 19 49

**Comment by pernielsentikaer** [comment]
We could do something like this:

2to6HpKWVcWAhAUQ

Then it's possible to bind alias or hotkey to each version, even disable a version, does it make any sense?

**Comment by ShubhamVerma1811** [comment]
I see.
But this would require to add the commands in the package.json
But the versions would be dynamic, coming from the github

File: extensions/nativebase-docs/package.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  "name": "nativebase-docs",
+  "title": "NativeBase Documention",
+  "description": "Search Nativebase documentation",
+  "icon": "nativebase-logo.png",
+  "author": "shubham",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "NativeBase Documention",
```
Comment (pernielsentikaer) [inline extensions/nativebase-docs/package.json:12]:
```suggestion
      "title": "Search Documention",
      "subtitle": "NativeBase",
```

File: extensions/nativebase-docs/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  if (!token) {
+    showNoTokenToast()
+  }
+
+  if (error) {
+    return <Detail markdown={`# Error: ${error.message}`} />
+  }
+
+  return (
+    <List navigationTitle="Select documentation version" isLoading={loading}>
```
Comment (pernielsentikaer) [inline extensions/nativebase-docs/src/index.tsx:73]:
```suggestion
    <List navigationTitle="Select Documentation Version" isLoading={loading}>
```

File: extensions/nativebase-docs/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      <List.Section title="Versions">
+        {versions.map((version, index) => (
```
Comment (pernielsentikaer) [inline extensions/nativebase-docs/src/index.tsx:75]: Shouldn't the list be “inverted” – showing the newest first?

File: extensions/nativebase-docs/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+          <List.Item
+            key={index}
+            title={version}
+            icon="../assets/nativebase-logo.png"
+            actions={
+              <ActionPanel>
+                <Action.Push
+                  title={`Open Docs for ${version}`}
```
Comment (pernielsentikaer) [inline extensions/nativebase-docs/src/index.tsx:83]: Could be visually a bit more awesome with an icon here too 😄

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @ShubhamVerma1811 👋 

Most of the documentation packages created at the moment is maintained manually, there is no way to programmatic create commands, but it's suggested in #939 – give your feedback in the thread 😄 

If you want to keep it as it is, then I think it's okay.

Did you consider removing the GitHub PAT in favor of cache?

I left some feedback for you to address 💪 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

File: extensions/nativebase-docs/src/components/Docs.tsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        method: 'GET'
+      }
+    )
+    const data = (await res.json()) as SidebarResponse
+    setSidebar(data?.sidebar)
+  }
+
+  return (
+    <List
+      navigationTitle={`NB Docs for version ${props.version}`}
```
Comment (pernielsentikaer) [inline extensions/nativebase-docs/src/components/Docs.tsx:30]:
```suggestion
      navigationTitle={`NativeBase ${props.version} Documentation`}
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @ShubhamVerma1811 👋 

Much better 🔥 Found one last thing before we're ready to merge 🚀 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by ShubhamVerma1811** [comment]
Sure!
Looking into how to cache the results.

**Comment by pernielsentikaer** [comment]
Awesome @ShubhamVerma1811 🔥

**Comment by ShubhamVerma1811** [comment]
Hey @pernielsentikaer 
Made the necessary changes!

**Comment by pernielsentikaer** [comment]
Awesome @ShubhamVerma1811 🔥 🔥 

Could you change in the README also, so it don't say anything about PAT? 😄

**Comment by ShubhamVerma1811** [comment]
> Awesome @ShubhamVerma1811 🔥 🔥
> 
> Could you change in the README also, so it doesn't say anything about PAT? 😄

Done!

**Review by pernielsentikaer — APPROVED** [review]
Hi @ShubhamVerma1811 👋 

Much better, thanks! 🔥

**Review by mattisssa — APPROVED** [review]

**Comment by mattisssa** [comment]
Hey @ShubhamVerma1811! Thanks for the contribution, you added some screenshots for your extension but the folder name you used was wrong, I've updated that but missed that the screenshots are actually doesn't match our [guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store#adding-screenshots). As far as I guess these are the screenshot sizes. 

Could you please take a look at that and update it?

**Comment by ShubhamVerma1811** [comment]
Yay! ❤️

**Pull request merged**